flink-clients/src/main/java/org/apache/flink/client/ClientUtils.java

```diff
@@ -144,4 +149,29 @@ public static void waitUntilJobInitializationFinished(
             throw new RuntimeException("Error while waiting for job to be initialized", throwable);
         }
     }
+
+    /**
+     * The client reports the heartbeat to the dispatcher for aliveness.
+     *
+     * @param jobClient The job client.
+     * @param interval The heartbeat interval.
+     * @param timeout The heartbeat timeout.
+     * @return The ScheduledExecutorService which reports heartbeat periodically.
+     */
+    public static ScheduledExecutorService reportHeartbeatPeriodically(
+            JobClient jobClient, long interval, long timeout) {
```

Comment:
It would be better to check that `timeout` is larger than `interval`, and if not fail with a descriptive error message.